src/core/lib/layer-manager.js

```diff
@@ -145,23 +171,21 @@ export default class LayerManager {
     Object.assign(this.context, parameters);
   }
 
+  // Update the view descriptor list and set change flag if needed
   setViewports(viewports) {
-    viewports = flatten(viewports, {filter: Boolean});
-
-    // Viewports are "immutable", so we can shallow compare
-    const oldViewports = this.context.viewports;
-    const viewportsChanged = viewports.length !== oldViewports.length ||
-      viewports.some((_, i) => viewports[i] !== oldViewports[i]);
-
-    if (viewportsChanged) {
-      this._needsRedraw = 'Viewport changed';
-
-      // Need to ensure one viewport is activated
-      const viewport = viewports[0];
-      assert(viewport instanceof Viewport, 'Invalid viewport');
-
-      this.context.viewports = viewports;
-      this._activateViewport(viewport);
+    // Ensure viewports are wrapped in descriptors
+    const viewDescriptors = flatten(viewports, {filter: Boolean})
+      .map(viewport => viewport instanceof Viewport ? {viewport} : viewport);
+
+    this.viewDescriptorsChanged =
+      this.viewDescriptorsChanged ||
```

Comment:
>this.viewDescriptorsChanged = this.viewDescriptorsChanged || ...

`this.viewDescriptorsChanged` is will be false here always, can be removed from this equation.